packages/backend/src/accounting/psql/service.ts

```diff
@@ -160,3 +227,26 @@ async function postAccountWithdrawal(
 ): Promise<void | TransferError> {
   throw new Error('Not implemented')
 }
+
+async function getAccountBalances(
+  deps: ServiceDependencies,
+  account: LedgerAccount
+): Promise<AccountBalance> {
+  const { credits, debits } =
+    await deps.ledgerTransferService.getAccountTransfers(account.id)
+
+  return {
+    creditsPosted: sum(credits, LedgerTransferState.POSTED),
+    creditsPending: sum(credits, LedgerTransferState.PENDING),
+    debitsPosted: sum(debits, LedgerTransferState.POSTED),
+    debitsPending: sum(debits, LedgerTransferState.PENDING)
+  }
+}
+
+function sum(transfers: LedgerTransfer[], state: LedgerTransferState) {
+  return transfers
+    .filter((transfer) => transfer.state === state)
+    .reduce((sum, transfer) => {
+      return sum + transfer.amount
+    }, 0n)
```

Comment:
Would it be worth trying to reduce this to one passthrough each for `credits` and `debits`?

Reply:
Yeah, I can do a single passthrough. I went with this since it seemed like it was pretty easy to read/glance but traded out perf